src/lib/amd.c

```diff
@@ -125,13 +126,22 @@ int amd_em_enabled(const char *path, struct led_ctx *ctx)
 		amd_ipmi_platform = AMD_PLATFORM_DAYTONA_X;
 	}
 
+	/* Universal to nvme led ctl */
+	if (using_amd_nvme_slot) {
+		amd_interface = AMD_INTF_NVME_SLOT;
+		amd_ipmi_platform = AMD_PLATFORM_UNSET;
+	}
```

Comment:
I have a question about the introduction of the new command option to specify using the nvme slot capabilities. You provide a function to determine if this is a  nvme controller, is_nvme_controller(), that is used in checking for enclosure management enabled and other routines. If the is_nvme_controller() call fails an error will be returned even if the new option is specified.

Is there a reason we can't just use the is_nvme_controller() function to determine if we should use nvme slot control methods instead of requiring a user to specify this with the new option? At this point you could set amd_interface and amd_ipmi_platform based on the call to is_nvme_controller().